src/main/java/org/jenkinsci/plugins/github_branch_source/Endpoint.java

```diff
@@ -89,29 +99,37 @@ public int hashCode() {
     @Extension
     public static class DesciptorImpl extends Descriptor<Endpoint> {
 
+        private static final Logger LOGGER = Logger.getLogger(DesciptorImpl.class.getName());
+
         @Override
         public String getDisplayName() {
             return "";
         }
 
-        public FormValidation doCheckApiUrl(@QueryParameter String value) {
-            if (Util.fixEmptyAndTrim(value) == null) {
-                return FormValidation.error("You must specify the API URI");
+        @Restricted(NoExternalUse.class)
+        public FormValidation doCheckApiUri(@QueryParameter String apiUri) {
+            if (Util.fixEmptyAndTrim(apiUri) == null) {
+                return FormValidation.warning("You must specify the API URL");
             }
-            Pattern enterprise = Pattern.compile("^https?://(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*"
-                    + "([A-Za-z]|[A-Za-z][A-Za-z0-9\\-]*[A-Za-z0-9])/api/v3/?$");
-            if (!enterprise.matcher(value).matches()) {
-                return FormValidation.warning(
-                        "This does not look like a GitHub Enterprise API URI. Expecting something like "
-                                + "https://[hostname or ip]/api/v3/");
+            try {
+                URL api = new URL(apiUri);
+                GitHub github = GitHub.connectToEnterpriseAnonymously(api.toString());
+                github.checkApiUrlValidity();
+                return FormValidation.ok();
+            } catch (MalformedURLException mue) {
```

Comment:
I know but we are not using this [feature](http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html) of Java SE 7 in this plugin.